signac/contrib/project.py

```diff
@@ -120,7 +134,8 @@ class Project(object):
     def __init__(self, config=None):
         if config is None:
             config = load_config()
-        self._config = config
+        self._config = ProjectConfig(config)
+        self._config._mutable = False
```

Comment:
You should be able to avoid manipulating the flag like this by rearranging the constructor calls as outlined above.